packages/aila/src/protocol/jsonPatchProtocol.ts

```diff
@@ -527,11 +535,25 @@ function tryParseJson(str: string): {
     }
     return { parsed, isPartial: false };
   } catch (e) {
-    // If parsing fails, assume it's partial
-    return { parsed: null, isPartial: true };
+    try {
```

Comment:
This now allows us to use process patches that are part of the llmMessage even though the llmMessage itself is still streaming in

Reply:
I assume that this will make the parsed value change on every streamed token. Would this result in a new render, or does it go into some logic that condenses it down into a value we care about?

Reply:
Yes, I've now added logic that reduces the parsing down significantly - we only parse once a section is detected as complete.